tests/component-plugin/src/main/java/org/mule/functional/api/component/EqualsLogChecker.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.functional.api.component;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static org.mule.runtime.api.exception.MuleException.EXCEPTION_MESSAGE_DELIMITER;
+import static org.mule.runtime.core.api.util.StringUtils.EMPTY;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class EqualsLogChecker extends AbstractLogChecker {
+
+  private String expectedLogMessage = EMPTY;
+  private boolean shouldFilterLogMessage;
+
+
+  @Override
+  public void check(String logMessage) {
+    StringBuilder errors = new StringBuilder();
+    List<String> expectedLines = splitLines(expectedLogMessage);
+    List<String> actualLines = getMessageLinesFromLogLines(splitLines(logMessage));
+
+    checkLineCount(expectedLines, actualLines, errors);
+
+    compareLines(expectedLines, actualLines, errors);
+
+    String errorMessage = errors.toString();
+    if (!StringUtils.isBlank(errorMessage)) {
+      throw new AssertionError(lineSeparator() + errorMessage);
+    }
+  }
+
+  private void checkLineCount(List<String> expectedLog, List<String> actualLog, StringBuilder errorCatcher) {
+    if (expectedLog.size() != actualLog.size()) {
+      errorCatcher.append(format("Log lines differs from expected one. It has %d lines and it's expecting %d\n",
+                                        actualLog.size(), expectedLog.size()));
+      errorCatcher.append(lineSeparator());
+    }
+  }
+
+  private void compareLines(List<String> expectedLogLines, List<String> actualLogLines, StringBuilder errorCatcher) {
+    int i;
+    for (i = 0; i < expectedLogLines.size(); i++) {
+      if (i >= actualLogLines.size()) {
+        errorCatcher.append(format("Missing expected line[%d]: %s\n", i, expectedLogLines.get(i)));
+        errorCatcher.append(lineSeparator());
+      } else {
+        if (!(expectedLogLines.get(i).trim().equals(actualLogLines.get(i).trim()))) {
+          errorCatcher.append(format("Difference found in line %d: \nEXPECTED: %s\nFOUND: %s\n", i,
```

Comment:
maybe you can use hamcrest to do this comparisons and generate the failure messages